src/server/bg_workers.js

```diff
@@ -72,18 +71,7 @@ function register_rpc() {
     });
 }
 
-
-function register_bg_worker(worker, run_batch_function) {
-    dbg.log0('Registering', worker.name, 'bg worker');
-    if (run_batch_function) {
-        worker.run_batch = run_batch_function;
-    }
-    if (!worker.name || !_.isFunction(worker.run_batch)) {
-        console.error('Name and run function must be supplied for registering bg worker', worker.name);
-        throw new Error('Name and run function must be supplied for registering bg worker ' + worker.name);
-    }
-    background_scheduler.run_background_worker(worker);
-}
+const register_bg_worker = background_scheduler.register_bg_worker;
```

Comment:
this will not work. this function is not bound to its instance so calling it will fail to resolve `this`.
You could replace with -
```suggestion
const register_bg_worker = 
    (worker, run_batch_function) => 
    background_scheduler.register_bg_worker(worker, run_batch_function)
```